File: 2026-03/src/day_11/boj_18937/Main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int nimsum = 0;

    int n;
    cin >> n;

    while (n--) {
        int x;
        cin >> x;
        nimsum ^= x - 2;
    }

    string s;
    cin >> s;
    bool is_white_king_name = s == "Whiteking";

    if (nimsum) {
        if (is_white_king_name) {
            cout << "Whiteking";
        } else {
            cout << "Blackking";
        }
    } else {
        if (is_white_king_name) {
            cout << "Blackking";
        } else {
            cout << "Whiteking";
        }
    }
}

File: 2026-03/src/day_11/boj_18937/Main.java
package day_11.boj_18937;

import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;

        int nimsum = 0;

        int n = Integer.parseInt(br.readLine());
        st = new StringTokenizer(br.readLine());
        while (n-- > 0) {
            int x = Integer.parseInt(st.nextToken());
            nimsum ^= x - 2;
        }
        boolean isWhiteKingName = br.readLine().equals("Whiteking");

        if (nimsum != 0) {
            if (isWhiteKingName) {
                System.out.println("Whiteking");
            } else {
                System.out.println("Blackking");
            }
        } else {
            if (isWhiteKingName) {
                System.out.println("Blackking");
            } else {
                System.out.println("Whiteking");
            }
        }
    }
}
